### ISSUE TITLE
Data read from .rec are not shuffled

### ISSUE THREAD

**Issue by youhangtian** [NVIDIA/DALI#348]
I use mxnet with dali, but find that data read from .rec are not shuffled, even though I set 'random_shuffle = True' in 'ops.MXNetReader'. Also, the train loss decrease at first, but will increase to the beginning after some epochs. Anyone face the same problem?

**Comment by JanuszL** [comment]
Hi,
I just tested adding `print` in PyTorch example from https://github.com/NVIDIA/DALI/pull/343 and commenting out all but MXNetReader and it works well for me.
Could you provide some minimal repro script that confirms that issue on your side?

**Comment by youhangtian** [comment]
I add the code in incubator-mxnet/python/mxnet/module/base_module.py to print the label as bellow:
        ################################################################################
        # training loop
        ################################################################################
        for epoch in range(begin_epoch, num_epoch):
            tic = time.time()
            eval_metric.reset()
            nbatch = 0
            data_iter = iter(train_data)
            end_of_batch = False
            next_data_batch = next(data_iter)
            while not end_of_batch:
                data_batch = next_data_batch
                if monitor is not None:
                    monitor.tic()
                self.forward_backward(data_batch)
                self.update()

                print data_batch
                print data_batch[0].provide_label
                print data_batch[0].label

And then run the imagenet example two times. The labels printed are the same 
![2018-12-07 21 50 50](https://user-images.githubusercontent.com/5603914/49651770-183e9780-fa6b-11e8-9892-51646643e983.png)
![2018-12-07 21 52 11](https://user-images.githubusercontent.com/5603914/49651792-255b8680-fa6b-11e8-982d-fe327d1649aa.png)

**Comment by mzient** [comment]
If you're referring to multiple runs of the entire training cycle, it's actually expected to produce the same (pseudo-random) shuffle each time. Is the order the same as if there was no random_shuffle at all?

**Comment by JanuszL** [comment]
If you set seed during the [pipeline](https://docs.nvidia.com/deeplearning/sdk/dali-developer-guide/docs/api.html#nvidia.dali.pipeline.Pipeline) creation it is expected to get 100% reproducibility. If you are not setting it seed its will be taken form the system time and it will vary from run to run.
